### PR TITLE
Correct "Build from Private Repo" Doc

### PR DESCRIPTION
**Pull request by paolaperaza**
Addresses: https://github.com/astronomer/issues/issues/1531. Related: https://github.com/astronomer/issues/issues/1386

Hey all, took a look at some needed corrections in this doc based on a report from Pocket (Cloud Customer).

A few things:

- Updated example `Dockerfile.build` file that previously referenced an 0.7.5 Airflow Image to our most recent 1.10.10 Alpine AC Image
- Updated Instructions for a customer to just add `FROM custom-ap-airflow` to their Dockerfile with guidance to set `FROM custom-ap-airflow:<airflow-image>`
- Added link to Airflow Versioning Doc

@schnie @adam2k Haven't tried this myself so not 100% sure on what else may need to be updated here. Can you guys take a look? Quote from Customer:

> We figured this out on our own last week by walking through your GitHub repo.
> 
> The problem is that the CLI tool assumes every image is a <tag>:<version> pair, when in fact the version is optional and defaults to “latest”.
> By updating the Dockerfile to read `FROM custom-ap-airflow:1.10.7-buster-onbuild` instead of `FROM custom-ap-airflow` — which was taken verbatim from the example at https://www.astronomer.io/docs/build-from-private-repo/ 
> 
> You should update both the docs and/pr the GitHub code, because taken together, they do not work.

Zendesk Ticket for reference: https://astronomer.zendesk.com/agent/tickets/1996

### CODE REVIEW

**Comment by paolaperaza** [comment]
One specific question I have. This is part of the instructions for Step 2 (Build your Image).

> Run the following in your terminal:

```
$ docker build -f Dockerfile.build --build-arg PRIVATE_RSA_KEY="$(cat ~/.ssh/id_rsa)" -t custom-ap-airflow
```

If the image they're gonna reference in their Dockerfile is indeed `FROM custom-ap-airflow:1.10.10-alpine3.10` (for example), does the end of this command need to get updated?

**Comment by adam2k** [comment]
@paolaperaza I can confirm the issue:

Dockerfile:

```
FROM astronomerinc/ap-airflow
```

Attempt to deploy:

```
🚀  astro-cli-playground astro deploy
Authenticated to staging.astronomer.io

Select which airflow deployment you want to deploy to:
 #     LABEL                    DEPLOYMENT NAME        WORKSPACE        DEPLOYMENT ID
 1     test-payload-metrics     opaque-corona-9917     Adam Staging     ckcnoy65i24z80j95g5x4g3h0

> 1
Deploying: opaque-corona-9917
opaque-corona-9917/airflow
Building image...
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
github.com/astronomer/astro-cli/docker.GetImageTagFromParsedFile(0xc000428380, 0x1, 0x1, 0x1, 0x1, 0x0, 0x0)
	/Users/andrii/workspace/src/github.com/astronomer/astro-cli/docker/parse.go:106 +0x19e
github.com/astronomer/astro-cli/airflow.Deploy(0xc000020044, 0x32, 0x0, 0x0, 0xc0004b4340, 0x19, 0x15e5900, 0xc00040c0c8, 0x18552c0)
	/Users/andrii/workspace/src/github.com/astronomer/astro-cli/airflow/docker.go:599 +0x8a1
github.com/astronomer/astro-cli/cmd.deploy(0xc0004c7180, 0x1d37f90, 0x0, 0x0, 0x0, 0x0)
	/Users/andrii/workspace/src/github.com/astronomer/astro-cli/cmd/deploy.go:73 +0xfe
github.com/astronomer/astro-cli/vendor/github.com/spf13/cobra.(*Command).execute(0xc0004c7180, 0x1d37f90, 0x0, 0x0, 0xc0004c7180, 0x1d37f90)
	/Users/andrii/workspace/src/github.com/astronomer/astro-cli/vendor/github.com/spf13/cobra/command.go:826 +0x453
github.com/astronomer/astro-cli/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc00049a500, 0x18552c0, 0xc00000e018, 0xc00049a500)
	/Users/andrii/workspace/src/github.com/astronomer/astro-cli/vendor/github.com/spf13/cobra/command.go:914 +0x2fb
github.com/astronomer/astro-cli/vendor/github.com/spf13/cobra.(*Command).Execute(...)
	/Users/andrii/workspace/src/github.com/astronomer/astro-cli/vendor/github.com/spf13/cobra/command.go:864
main.main()
	/Users/andrii/workspace/src/github.com/astronomer/astro-cli/main.go:17 +0xe1
```

Putting back my Dockerfile to the default that was generated by the CLI: 
```
FROM astronomerinc/ap-airflow:1.10.5-alpine3.10-onbuild
```

```
🚀  astro-cli-playground astro deploy
Authenticated to staging.astronomer.io

Select which airflow deployment you want to deploy to:
 #     LABEL                    DEPLOYMENT NAME        WORKSPACE        DEPLOYMENT ID
 1     test-payload-metrics     opaque-corona-9917     Adam Staging     ckcnoy65i24z80j95g5x4g3h0

> 1
Deploying: opaque-corona-9917
opaque-corona-9917/airflow
Building image...
Sending build context to Docker daemon  11.78kB
Step 1/1 : FROM astronomerinc/ap-airflow:1.10.5-alpine3.10-onbuild
# Executing 5 build triggers
 ---> Using cache
 ---> Using cache
 ---> Using cache
 ---> Using cache
 ---> Using cache
 ---> d69896474a38
Successfully built d69896474a38
Successfully tagged opaque-corona-9917/airflow:latest
Pushing image to Astronomer registry
The push refers to repository [registry.staging.astronomer.io/opaque-corona-9917/airflow]
619b085959a8: Pushed
efe888097342: Pushed
f4549897ba0d: Pushed
45547edd6eb2: Pushed
1c69cf2507ce: Pushed
10ec0c9f0c27: Pushed
93299e429073: Pushed
1eb2db5d6fda: Pushed
c5cee01c0338: Pushed
d3048a87cdd6: Pushed
af0bae2174d3: Pushed
6637f7a62625: Pushed
1f839a184cac: Pushed
1b3ee35aacca: Pushed
deploy-1: digest: sha256:3ef6079883e0d1aa5ac04f882b65acad2a043adaefd0fc7e7acdb696b368624a size: 3230
Untagged: registry.staging.astronomer.io/opaque-corona-9917/airflow:deploy-1
Untagged: registry.staging.astronomer.io/opaque-corona-9917/airflow@sha256:3ef6079883e0d1aa5ac04f882b65acad2a043adaefd0fc7e7acdb696b368624a
Deploy succeeded!
```

I know that @andriisoldatenko wanted to make some additional changes to error handle this on the CLI side too, but the docs look good now.

**Comment by adam2k** [comment]
> One specific question I have. This is part of the instructions for Step 2 (Build your Image).
> 
> > Run the following in your terminal:
> 
> ```
> $ docker build -f Dockerfile.build --build-arg PRIVATE_RSA_KEY="$(cat ~/.ssh/id_rsa)" -t custom-ap-airflow
> ```
> 
> If the image they're gonna reference in their Dockerfile is indeed `FROM custom-ap-airflow:1.10.10-alpine3.10` (for example) instead of the previous `FROM custom-ap-airflow`, does the end of this command need to get updated?

To close the loop on this.  I did some testing with @paolaperaza and we needed to update the command to: `docker build -f Dockerfile.build --build-arg PRIVATE_RSA_KEY="$(cat ~/.ssh/id_rsa)" -t custom-ap-airflow:1.10.10-alpine3.10 .`

I also needed to modify my python version to get the Dockerfile.build to run correctly.

File: v0.16/build-from-private-repo.md [hunk 1]
```diff
@@ -80,15 +80,21 @@ Now, let's build a Docker image based on the requirements above that we'll then
 Run the following in your terminal:
 
 ```
-$ docker build -f Dockerfile.build --build-arg PRIVATE_RSA_KEY="$(cat ~/.ssh/id_rsa)" -t custom-ap-airflow
+$ docker build -f Dockerfile.build --build-arg PRIVATE_RSA_KEY="$(cat ~/.ssh/id_rsa)" -t custom-<airflow-image> .
```
Comment (adam2k) [inline v0.16/build-from-private-repo.md:83]: Should this be 
```Suggestion
$ docker build -f Dockerfile.build --build-arg PRIVATE_RSA_KEY="$(cat ~/.ssh/id_rsa)" -t custom-<airflow-image>:<version> .
```

Comment (paolaperaza) [reply v0.16/build-from-private-repo.md:83]: @adam2k Hm I think this is always encapsulated in `<airflow-image`> which I tried to clarify in the instructions + example.

E.g.

> If you're running `astronomerinc/ap-airflow:1.10.10-alpine3.10` as specified above, this line would read:
> 
> ```
> FROM custom-ap-airflow:1.10.10-alpine3.10
> ```

Think that's clear enough?

Comment (adam2k) [reply v0.16/build-from-private-repo.md:84]: ah okay.  Yeah, if that hasn't caused any customer issues then let's leave it alone.

File: v0.16/build-from-private-repo.md [hunk 2]
```diff
@@ -58,9 +69,10 @@ In 3 stages, this file is bundling up your SSH keys, OS-Level packages in `packa
 
 A few notes:
 - The `Private RSA Key` = [SSH Key generated via GitHub](https://help.github.com/en/articles/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)
-- Make sure to replace the first line of this file (`FROM`..) with your _current_ Docker image
+- Make sure to replace the first line of this file (`FROM`..) with your Airflow Image (Step 2 above)
 - If you don't want keys in this file to be pushed back up to your GitHub repo, consider adding this file to `.gitignore`
 - Make sure your custom OS-Level packages are in `packages.txt` and your Python packages in `requirements.txt` within your repo
+- If you're running Python 3.7 on your machine, replace the reference to Python 3.6 under `## # Copy requirements directory` with `/usr/lib/python3.7/site-packages/` above
```
Comment (adam2k) [inline v0.16/build-from-private-repo.md:75]: 👍

**Comment by adam2k** [comment]
The updates look good to me!  Thanks.

**Pull request merged**